kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/SchemaUtils.java

```diff
@@ -0,0 +1,375 @@
+                break;
+              case "day":
+              case "days":
+                specBuilder.day(matcher.group(2));
+                break;
+              case "hour":
+              case "hours":
+                specBuilder.hour(matcher.group(2));
+                break;
+              case "bucket":
+                {
+                  Pair<String, Integer> args = transformArgPair(matcher.group(2));
+                  specBuilder.bucket(args.first(), args.second());
+                  break;
+                }
+              case "truncate":
+                {
+                  Pair<String, Integer> args = transformArgPair(matcher.group(2));
+                  specBuilder.truncate(args.first(), args.second());
+                  break;
+                }
+              default:
+                throw new UnsupportedOperationException("Unsupported transform: " + transform);
+            }
+          } else {
+            specBuilder.identity(partitionField);
+          }
+        });
+    return specBuilder.build();
+  }
+
+  private static Pair<String, Integer> transformArgPair(String argsStr) {
+    List<String> parts = Splitter.on(',').splitToList(argsStr);
+    if (parts.size() != 2) {
+      throw new IllegalArgumentException("Invalid argument " + argsStr + ", should have 2 parts");
+    }
+    return Pair.of(parts.get(0).trim(), Integer.parseInt(parts.get(1).trim()));
+  }
+
+  public static Type toIcebergType(Schema valueSchema, IcebergSinkConfig config) {
+    return new SchemaGenerator(config).toIcebergType(valueSchema);
+  }
+
+  public static Optional<Type> inferIcebergType(Object value, IcebergSinkConfig config) {
+    return new SchemaGenerator(config).inferIcebergType(value);
+  }
+
+  static class SchemaGenerator {
+
+    private int fieldId = 1;
+    private final IcebergSinkConfig config;
+
+    SchemaGenerator(IcebergSinkConfig config) {
+      this.config = config;
+    }
+
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
+    Type toIcebergType(Schema valueSchema) {
+      switch (valueSchema.type()) {
+        case BOOLEAN:
+          return BooleanType.get();
+        case BYTES:
+          if (valueSchema.name() != null && valueSchema.name().equals(Decimal.LOGICAL_NAME)) {
+            int scale = Integer.parseInt(valueSchema.parameters().get(Decimal.SCALE_FIELD));
+            return DecimalType.of(38, scale);
+          }
+          return BinaryType.get();
+        case INT8:
+        case INT16:
+          return IntegerType.get();
+        case INT32:
+          if (valueSchema.name() != null) {
+            if (valueSchema.name().equals(Date.LOGICAL_NAME)) {
+              return DateType.get();
+            } else if (valueSchema.name().equals(Time.LOGICAL_NAME)) {
+              return TimeType.get();
+            }
+          }
+          return IntegerType.get();
+        case INT64:
+          if (valueSchema.name() != null && valueSchema.name().equals(Timestamp.LOGICAL_NAME)) {
```

Comment:
Nit: prefer String constants on the left side to avoid needing an extra null check. `"x".equals(null)` will return false. That would simplify a few of the cases in this method.

Reply:
Yes, thanks, I updated these